### PR TITLE
Ep 761 - Update front - implementation page

### PR DESCRIPTION
**Pull request by rebeccadumazert**
## 🎯 Objectif

@renardpal updated the model following user-tests 👉 [JUST HERE ](https://www.figma.com/file/rGcud9RbwBSmwXzyhMOnvp/Espace-Partenaire?type=design&node-id=136-1697&mode=design&t=xDJMjrhAI3OxppHc-0)👈

### CODE REVIEW

File: front/src/providers/documentation/DocumentationSideMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            },
+            {
+              isActive: currentAnchor == '#url' ? true : false,
+              linkProps: {
+                to: '#requirements',
+              },
+              text: 'Connaissances préalables',
+            },
+            {
+              isActive: currentAnchor == '#scopes' ? true : false,
```
Comment (douglasduteil) [inline front/src/providers/documentation/DocumentationSideMenu.tsx:37]: **note:** the test is already a boolean

```suggestion
              isActive: currentAnchor == '#scopes',
```

File: front/src/providers/documentation/EspaceDocumentation.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { DocumentationContent } from './DocumentationContent';
+import { DocumentationSideMenu } from './DocumentationSideMenu';
+
+export const EspaceDocumentation = () => {
+  return (
+    <div className="fr-container--fluid fr-mt-10v">
+      <div className="fr-grid-row">
+        <DocumentationSideMenu></DocumentationSideMenu>
```
Comment (douglasduteil) [inline front/src/providers/documentation/EspaceDocumentation.tsx:8]: **suggestion:** Components without children could be self-closing tag
```suggestion
        <DocumentationSideMenu />
```

File: front/src/providers/documentation/EspaceDocumentation.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        <div className="fr-col-12 fr-col-md">
+          <DocumentationContent></DocumentationContent>
```
Comment (douglasduteil) [inline front/src/providers/documentation/EspaceDocumentation.tsx:10]: **suggestion:** Components without children could be self-closing tag

```suggestion
          <DocumentationContent />
```

File: front/src/providers/list/EspaceList.tsx [hunk 4]
```diff
@@ -1,23 +1,24 @@
-import React from 'react';
+// import React from 'react';
```
Comment (douglasduteil) [inline front/src/providers/list/EspaceList.tsx:1]: **question:** What are those comments for ?

File: front/src/providers/list/EspaceList.tsx [hunk 5]
```diff
@@ -1,23 +1,24 @@
...
 import { useState } from 'react';
-import { backendClient } from '../../clients/back-client';
+// import { backendClient } from '../../clients/back-client';
```
Comment (douglasduteil) [inline front/src/providers/list/EspaceList.tsx:3]: **question:** What are those comments for ?

File: front/src/providers/list/EspaceList.tsx [hunk 6]
```diff
@@ -1,23 +1,24 @@
...
+  //     try {
+  //       const data = await backendClient.getList();
+  //       console.log(data);
+  //       setList(data);
+  //     } catch (error) {
+  //       console.error('Error fetching data:', error);
+  //     }
+  //   }
+  //   fetchData();
+  // }, []);
```
Comment (douglasduteil) [inline front/src/providers/list/EspaceList.tsx:21]: **question:** What are those comments for ?

**Review by douglasduteil — APPROVED** [review]
Looks good.

I don't think you should keep already committed code as comments Git will remember the code you wrote :)

<p align="center">
    
</p>

File: front/src/providers/details/KeyProductionData.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+export const KeyProductionData: React.FC<KeyProductionDataProps> = ({
+  oidcClientForm,
+}) => {
+  return (
+    <div className="fr-mb-10v">
+      <Title2 title="Clés des données de production" id="save" />
+      <p>
+        Pour accéder aux données de production, il vous faut une habilitation
+        juridique sur Datapass. Vous recevrez vos clés de production dès que
+        l’habilitation sera validée (XX jours en moyenne).
```
Comment (gdarquie) [inline front/src/providers/details/KeyProductionData.tsx:19]: "XX jours" Don't forget to change that one day

File: front/src/providers/details/KeyProductionData.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+      </p>
+      <p>
+        Le membre de votre équipe qui représente officiellement l’organisation
+        doit remplir l’habilitation.Il s’agit souvent d’un agent public,
```
Comment (gdarquie) [inline front/src/providers/details/KeyProductionData.tsx:23]: You need a space after "l’habilitation."

File: front/src/providers/details/ProviderDetails.tsx [hunk 9]
```diff
@@ -31,38 +30,44 @@ export function ProviderDetails() {
...
-            Lire la documentation
-          </Button>
+        <Title1 title="Implémentation" />
+        <div className="fr-container--fluid">
+          <div className="fr-grid-row fr-grid-row--gutters">
+            <CardInfos
+              title="Faites vos tests"
+              stringArray={[
+                'Accédez à des clés de test',
+                'Configurer vos URLS de login et de callback',
```
Comment (gdarquie) [inline front/src/providers/details/ProviderDetails.tsx:40]: Configurer => Configurez

File: front/src/providers/details/ProviderScope.tsx [hunk 10]
```diff
@@ -37,7 +37,7 @@ export const ProviderScope = () => {
             Vous pouvez utiliser tous les paramètres disponibles via
             MonComptePro pendant vos tests. Gardez en mémoire que pour que votre
             habilitation soit validée, vous devrez utiliser uniquement
-            <b>les données strictement nécessaires</b> à votre service.
+            <b> les données strictement nécessaires</b> à votre service.
```
Comment (gdarquie) [inline front/src/providers/details/ProviderScope.tsx:40]: not important but the space should be outside the b

File: front/src/providers/documentation/DocumentationContent.tsx [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        </li>
+        <li>Mon implémentation fonctionne</li>
+        <li>
+          J’ai récupéré mon client_id et mon client_secret de production en
+          demandant à l'équipe MonComptePro de me les remettre de manière
+          sécurisée
+        </li>
+        <li>Mise en production 🚀</li>
+      </ul>
+      <h1 id="requirements">Connaissances préalables</h1>
```
Comment (gdarquie) [inline front/src/providers/documentation/DocumentationContent.tsx:59]: Many h1 on the same page? Possible to use h2?

File: front/src/providers/details/ProviderKey.tsx [hunk 12]
```diff
@@ -49,47 +54,71 @@ export const ProviderKey = () => {
...
+      )}
+      <p className="fr-mt-10v">
+        <b>S'identifier pour sauvergarder les clés</b>
+      </p>
+      <p>✅ la durée de vie des clés deviendra illimitée</p>
+      <p>✅ vous pourrez partager vos clés aux membres de vos équipes</p>
+      <Button
+        iconId="fr-icon-save-3-fill"
+        disabled
+        onClick={function noRefCheck() {}}
```
[the remark this answers is not in the document]
Comment (rebeccadumazert) [reply front/src/providers/details/ProviderKey.tsx:117]: it was a technique that @BenoitSerrano and I came up with to help us make progress on the model without having a clickable button that doesn't take any action !

File: front/src/cards/CardInfos.tsx [hunk 13]
```diff
@@ -0,0 +1,99 @@
+import Button from '@codegouvfr/react-dsfr/Button';
+import { COLORS } from '../constants';
+import { FrIconClassName } from '@codegouvfr/react-dsfr';
+// import { Card } from '@codegouvfr/react-dsfr/Card';
+
+interface CardInfosProps {
+  title: string;
+  stringArray: string[];
```
[the remark this answers is not in the document]
Comment (rebeccadumazert) [reply front/src/cards/CardInfos.tsx:8]: ahahahaha sorry for your eyes 🩸 tear of blood

File: front/src/custom-style.css [hunk 14]
```diff
@@ -0,0 +1,10 @@
+@media (min-width: 1023px){
+    .divider{
+        box-shadow: inset 0 -1px 0 0 var(--border-default-grey);
+    }
+}
+
+/* ADD PERSONALIZED BORDER TO BUTTON OF HEADER */
```
[the remark this answers is not in the document]
Comment (rebeccadumazert) [reply front/src/custom-style.css:7]: I'm not just hurting your eyes, I'm enjoying it. 💧 tear of joy

File: front/src/providers/details/KeyProductionData.tsx (hunk 7 above)
Comment (rebeccadumazert) [reply front/src/providers/details/KeyProductionData.tsx:19]: Do we know the average number of days?

File: front/src/providers/details/ProviderKey.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (rebeccadumazert) [reply front/src/providers/details/ProviderKey.tsx:117]: I delete the OnClick !

**Pull request merged**